common/web/keyboard-processor/src/keyboards/defaultLayouts.ts

```diff
@@ -11,16 +11,17 @@ namespace com.keyman.keyboards {
   export type KLS = {[layerName: string]: string[]};
 
   // The following types provide type definitions for the full JSON format we use for visual keyboard definitions.
-  export type ButtonClass = number|"0"|"1"|"2"|"3"|"4"|"5"|"6"|"7"|"8"|"9"|"10"
+  export type ButtonClass       =  0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10;
+  export type ButtonClassString = "0"|"1"|"2"|"3"|"4"|"5"|"6"|"7"|"8"|"9"|"10";
 
   export type LayoutKey = {
     "id"?: string,
     "text"?: string,
-    "sp"?: ButtonClass,
-    "width"?: string,
+    "sp"?: ButtonClass | ButtonClassString,
+    "width"?: string | number,
     "layer"?: string, // Key derives any modifiers from the value set here if specified, not the actual display layer.
     "nextlayer"?: string,
-    "pad"?: string,
+    "pad"?: string | number,
     "sk"?: LayoutKey[]
   }
 
```

Comment:
`LayoutRow.id` should probably be fixed as well at this time... compiler currently always emits a string for compat but the new, clean touch layout source spec requires a number, so we should be moving in that direction:

https://github.com/keymanapp/keyman/blob/df34c064c75f87708060d580d84104da051f21d4/common/schemas/keyman-touch-layout/keyman-touch-layout.clean.spec.json#L69-L73

Note, the legacy spec allows either string or number:

https://github.com/keymanapp/keyman/blob/df34c064c75f87708060d580d84104da051f21d4/common/schemas/keyman-touch-layout/keyman-touch-layout.spec.json#L65-L77